### PR TITLE
Added announcement page for front user

### PR DESCRIPTION
**Pull request by bestworks221**
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

### CODE REVIEW

File: app/Http/Controllers/Announcement/AnnouncementController.php [hunk 1]
```diff
@@ -98,8 +109,44 @@ public function store(AnnouncementRequest $request): JsonResponse
...
+        $announcement = AnnouncementModel::find($id);
+
+        if (! $announcement) {
+            return $this->result->notFound();
+        }
+
+        $announcement->status = $status;
+        $announcement->save();
+
+        $users = User::all();
```
Comment (ri-bulatao) [inline app/Http/Controllers/Announcement/AnnouncementController.php:129]: Should be in-app Notifications not Email

Comment (ri-bulatao) [reply app/Http/Controllers/Announcement/AnnouncementController.php:129]: The Emails is for the Admins for the Leaves

File: database/seeders/PositionSeeder.php [hunk 2]
```diff
@@ -19,6 +19,12 @@ public function run()
         /**
          * Generate User Positions.
          */
-        Position::factory()->count(5)->create();
+        $positions = ['Developers', 'Graphics', 'Marketing', 'Project Leads', 'Managers'];
```
Comment (ri-bulatao) [inline database/seeders/PositionSeeder.php:22]: Please create Enums for the Positions

Comment (ri-bulatao) [reply database/seeders/PositionSeeder.php:22]: Then use the way of the Role when creating Psitions

File: resources/js/store/modules/announcements.js [hunk 3]
```diff
@@ -78,6 +78,17 @@ export const actions = {
         })
     },
 
+    postAnnouncement({commit, dispatch}, payload) {
+        return new Promise((resolve, reject) => {
+            axios.put(route('announcements.post'), payload)
+                .then(res => {
+                    let response = res.data
+                    resolve(response)
```
Comment (ri-bulatao) [inline resources/js/store/modules/announcements.js:86]: use this `resolve(res.data)` Remove the initialization

**Review by ri-bulatao — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Announcement/AnnouncementController.php (hunk 1 above)
Comment (bestworks221) [reply app/Http/Controllers/Announcement/AnnouncementController.php:129]: Added Events

File: database/seeders/PositionSeeder.php (hunk 2 above)
Comment (bestworks221) [reply database/seeders/PositionSeeder.php]: Updated to Enums and the way of creating positions

File: resources/js/store/modules/announcements.js (hunk 3 above)
Comment (bestworks221) [reply resources/js/store/modules/announcements.js]: Removed the initializations when response is not used in the action

**Pull request merged**